### DISCUSSION TITLE
I2C issues on raspberry pi pico micropython scan() works readfrom_mem fails

### DISCUSSION THREAD

**Discussion by t43riz** [micropython/micropython#9983] [Using MicroPython] [answered]
I am trying to read data from the MLX90614 temp sensor connected to the raspberry pi pico everything looks connected ok. 
3.3v to VIN
GND-GND
SCL to pin 7
SDA to pin 6

```
    from machine import I2C, Pin,SoftI2C
    import ustruct
    i2c = I2C(0,sda=Pin(6), scl=Pin(7))
    i2.scan()

```
This returns `[90]`

  ```


      >>> i2c.scan()
    [90]
```

But when i try  to `readfrom_mem` I always get an error

```
>>> i2c.readfrom_mem(90,0x06,10)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 5] EIO
```
Any ideas on whats going wrong?

**Comment by scruss** [comment]
Have you tried the [MLX90614](https://awesome-micropython.com/#temperature-ir) driver instead?

**Comment by t43riz** [comment]
This can be closed figured it out just few minutes after posting

**Reply by scruss** [reply]
It might help if you documented what you did, then accepted that as an answer. This would 'close' this topic, and help other uses who might have the same problem

**Comment by t43riz** [comment, marked as answer]
```
from machine import I2C, Pin,SoftI2C
import ustruct
import time
sda = Pin(6)
scl = Pin(7)
i2c = I2C(1, sda = sda, scl = scl, freq=100000)
```


I had to set the frequency and it started working

**Comment by apn-tech** [comment]
its working thank you
